packages/mwp-core/src/util/launchDarkly.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+const HEADER_FASTLY_CLIENT_IP = 'fastly-client-ip';
+const HEADER_FASTLY_X_REGION = 'x-region';
+
+/**
+ * Populates a LaunchDarklyUser object with member and request properties.
+ *
+ * @param member  a member state object
+ * @param request a request object
+ */
+export const getLaunchDarklyUser = (
+	member: Object,
+	request: HapiRequest
+): LaunchDarklyUser => {
+	const { id, name, email, country } = member;
+
+	const ip = getRemoteIp(request);
+	const key = (id && id.toString()) || '0';
+	const anonymous = key === '0';
+	const custom = getCustomAttributes(request);
+
+	const user: LaunchDarklyUser = {
+		key,
+		anonymous,
+		custom,
+	};
+
+	if (ip) {
+		user.ip = ip;
+	}
+	if (name) {
+		user.name = name;
+	}
+	if (email) {
+		user.email = email;
+	}
+	if (country) {
+		user.country = country;
+	}
+
+	return user;
+};
+
+const getRemoteIp = (request: HapiRequest): ?string => {
+	const { headers, info, query } = request;
+	const fromQuery = query && query.__set_geoip && query.__set_geoip.toString();
+	const fromHeaders = headers && headers[HEADER_FASTLY_CLIENT_IP];
+	const fromInfo =
+		info && typeof info.remoteAddress === 'string'
+			? info.remoteAddress
+			: undefined;
+	return fromQuery || fromHeaders || fromInfo;
+};
+
+const getCustomAttributes = (
+	request: HapiRequest
+): LaunchDarklyUser$CustomAttributes => {
+	const { headers } = request;
+
+	const location = (headers && headers[HEADER_FASTLY_X_REGION]) || '';
+	const [country, region] = location
+		.split('/')
+		.map(value => (value ? value.toUpperCase() : value));
+
+	const custom: LaunchDarklyUser$CustomAttributes = {};
+
+	if (country) {
+		custom.RequestCountry = country;
```

Comment:
```
const requestCountry = country && {RequestCountry: country}
 const custom: LaunchDarklyUser$CustomAttributes = {
 ...requestCountry,
...requestRegion,
};
```

Reply:
```
string [1] is not an object.

     packages/mwp-core/src/util/launchDarkly.js
      66│ 	const requestCountry = country && { RequestCountry: country };
      67│ 	const requestRegion = region && { RequestRegion: region };
      68│
      69│ 	return {
      70│ 		...requestCountry,
      71│ 		...requestRegion,
      72│ 	};
      73│ };
      74│

     /tmp/flow/flowlib_3df97785/core.js
 [1] 323│     split(separator?: string | RegExp, limit?: number): Array<string>;
```

Reply:
However this works:
```
const requestCountry = country ? { RequestCountry: country } : {};
const requestRegion = region ? { RequestRegion: region } : {};

return {
	...requestCountry,
	...requestRegion,
};
```

The object expansion in the return statement works because `LaunchDarklyUser$CustomAttributes` is an inexact type, so flow doesn't care that the expansion can potentially add properties that are not explicitly defined in the type